src/08_example/02_handling_connections_concurrently.md

```diff
@@ -0,0 +1,34 @@
+# Handling Connections Concurrently
```

Comment:
The structure here makes sense (concurrency first, then parallelism). But the contents and examples use task spawn, which is already parallel. That's why you have to talk about "the main thread" and so on here, but ideally that could be deferred to the next section.

It would be excellent from an educational perspective if this section was able to simply use a single task with concurrency (e.g. through `for_each_concurrent`), and then you can dedicate the next section to talking about multithreading and parallelism. The code change with adding spawn would be minor, but the effect on the application is big, so it'd be mostly a section talking about how it works.

Reply:
I agree that I'd like to introduce concurrency before talking about multithreading. However, I think the concept of concurrency with only a single task is confusing. Even if we're not blocking the executor while using `for_each_concurrent`, we can still only process one connection at a time. Maybe this could be reframed as a section on avoiding blocking, and then a section on processing connections concurrently by adding parallelism. What do you think?